File: H-Index/II/H-IndexII.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        int left = 0, right = n - 1;
        while(left <= right) {
        	int mid = left + (right - left) / 2;
        	if(n - mid == citations[mid])
        		return n - mid;
        	else if(n - mid < citations[mid])
        		right = mid - 1;
        	else
        		left = mid + 1;
        }
        return n - (right + 1);
    }
};

File: H-Index/II/H-IndexII.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        N = len(citations)
        left = 0
        right = len(citations) - 1
        while left <= right:
            mid = left + (right - left) / 2
            if N - mid == citations[mid]:
                return N - mid
            elif N - mid < citations[mid]:
                right = mid - 1
            else:
                left = mid + 1
        return N - (right + 1)